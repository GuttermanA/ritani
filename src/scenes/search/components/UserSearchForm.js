import React, { Component } from "react";
import { Form } from "../../../components";

class UserSearchForm extends Component {
  state = {
    username: ""
  };

  handleChange = event => {
    const target = event.target;
    this.setState({ [target.name]: target.value }, () =>
      console.log(this.state)
    );
  };

  render() {
    return (
      <Form>
        <Form.Field
          name="username"
          placeholder="Github Username"
          type="text"
          onChange={this.handleChange}
        />
        <Form.Button>Search</Form.Button>
      </Form>
    );
  }
}

export default UserSearchForm;
