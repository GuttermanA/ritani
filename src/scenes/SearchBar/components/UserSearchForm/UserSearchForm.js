import React, { Component } from "react";
import { Form } from "components";

class UserSearchForm extends Component {
  render() {
    const { handleSubmit, handleChange, disabled, username } = this.props;
    return (
      <Form onSubmit={handleSubmit}>
        <Form.Label fieldName="username">
          Enter Github Username to Retrieve Followers
        </Form.Label>
        <Form.Field
          name="username"
          placeholder="Github Username"
          type="text"
          onChange={handleChange}
          value={username}
          required
        />
        <Form.Button disabled={disabled}>Search</Form.Button>
      </Form>
    );
  }
}

export default UserSearchForm;
