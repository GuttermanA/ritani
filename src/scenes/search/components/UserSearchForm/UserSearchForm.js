import React, { Component } from "react";
import { Form, Container } from "components";

class UserSearchForm extends Component {
  render() {
    const { handleSubmit, handleChange, disabled, error } = this.props;
    return (
      <Container className="menu fluid center container">
        <Form onSubmit={handleSubmit}>
          <Form.Field
            name="username"
            placeholder="Github Username"
            type="text"
            onChange={handleChange}
            required
          />
          <Form.Button className="button" disabled={disabled}>
            Search
          </Form.Button>
        </Form>
      </Container>
    );
  }
}

export default UserSearchForm;
