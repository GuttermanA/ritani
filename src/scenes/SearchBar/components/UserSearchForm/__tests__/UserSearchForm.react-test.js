import React from "react";
import renderer from "react-test-renderer";
import UserSearchForm from "../UserSearchForm";

describe("<UserSearchForm/>", () => {
  it("renders correctly", () => {
    const component = renderer.create(<UserSearchForm />);
    expect(component).toMatchSnapshot();
  });

  it(`it has a "Search" button`, () => {});
});
