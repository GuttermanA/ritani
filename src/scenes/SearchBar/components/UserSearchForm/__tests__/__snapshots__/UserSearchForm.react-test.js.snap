// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSearchForm/> renders correctly 1`] = `
<form
  as="form"
>
  <label
    className="form label"
    htmlFor="username"
  >
    Enter Github Username to Retrieve Followers
  </label>
  <input
    className="form field"
    name="username"
    placeholder="Github Username"
    required={true}
    type="text"
  />
  <button
    className="form button"
    type="submit"
  >
    Search
  </button>
</form>
`;
